experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -0,0 +1,341 @@
+###############################################################################
+#
+#   Coxeter Groups
+#
+###############################################################################
+
+struct WeylGroup <: CoxeterGroup
```

Comment:
you probably will want this to be attribute storing on the long run...
```suggestion
@attributes mutable struct WeylGroup <: CoxeterGroup
```

I think it would be useful to have a comment (or better: docstring) that briefly explains in how far you consider "Weyl group" as a special case of Coxeter group (i.e., what makes it different from "other" kinds of of Coxeter groups)

Reply:
`WeylGroup` is the implemtation of `CoxGrpMin` in https://github.com/ulthiel/CoxeterGroups.jl. Since this PR is quite large already, in a follow up PR I want to add `CoxGrpRec` renamed to `CoxeterGroupGeneric` (?) and offer a generic Coxeter group constructor from a Coxeter matrix. In my opinion there should then be references for specials cases like Weyl Groups (with specialized algorithms). In https://github.com/ulthiel/CoxeterGroups.jl/issues/1 `CoxeterGroups{T}` was suggested. I am open for suggestions also from @ulthiel.
Also note that an abstract type is necessary to break cyclic dependency between root systems and Weyl groups.

(Maybe I also misunderstood your comment.)

Reply:
Whenever I read "Weyl group" I think "finite Coxeter group". Maybe I am too much used to the algebraic groups setting. Kac uses Weyl group more generally as well. I don't know what's best. But note that CoxGrpMin refers to an _algorithm_ (namely using minimal roots). I somehow like(d) to make this clear. But again, I don't know what's best.

Reply:
2 days later I really think there should be separate types for the separate algorithms for computing in (some classes of) Coxeter groups and the types should also be called like that, as originally in CoxeterGroups.jl. So, I would prefer CoxGrpMin to WeylGroup. Also, there is CoxGrpSym in CoxeterGroups.jl which is extremely useful for type-A computations. This should go in here as well. There may be CoxGrpPerm for a permutation group algorithm etc. The generic function coxeter_group(...) should select the type that's most appropriate for the given type. @fingolfin, what do you think?

Reply:
I will look at this next week.

Reply:
Could this be looked at before merging?

Reply:
To me Weyl group applies to infinite groups as well, in Kac Moody theory at least we consistently use that.

Having different types for different algorithms makes some sense, but there are also many drawbacks. But I don't think it's necessary to hold up this PR about it, as this is purely for the *experimental* branch.

If we want to make progress on this, I don't think discussions on PRs are the way forward. It just doesn't work that well. We need to sit together in a room with enough time (possible organize a little workshop) and hash it out... but in the end, actual code trumps all. If something is implemented and works, great; if it doesn't work, we can iterate on it.

I.e., "use case drive design" -- which is why I said in a PR comment elsewhere that I don't want to hold up this PR right now as I just don't have time to work on it or one code using it, and if it works for you guys right now, great. I already *know* I will need changes for my use cases (e.g. infinite root systems should not store explicit vectors of positive roots; and I need a `RootDatum` type; etc.) but all of that can (and IMHO: should) be added *later*.